runtime/Cpp/runtime/src/atn/ParserATNSimulator.cpp

```diff
@@ -697,24 +697,22 @@ std::vector<Ref<SemanticContext>> ParserATNSimulator::getPredsForAmbigAlts(const
 
 std::vector<dfa::DFAState::PredPrediction *> ParserATNSimulator::getPredicatePredictions(const antlrcpp::BitSet &ambigAlts,
   std::vector<Ref<SemanticContext>> altToPred) {
-  std::vector<dfa::DFAState::PredPrediction*> pairs;
-  bool containsPredicate = false;
   for (size_t i = 1; i < altToPred.size(); i++) {
-    Ref<SemanticContext> pred = altToPred[i];
+    const Ref<SemanticContext>& pred = altToPred[i];
 
     // unpredicted is indicated by SemanticContext.NONE
     assert(pred != nullptr);
 
-    if (ambigAlts.test(i)) {
-      pairs.push_back(new dfa::DFAState::PredPrediction(pred, (int)i)); /* mem-check: managed by the DFAState it will be assigned to after return */
-    }
     if (pred != SemanticContext::NONE) {
-      containsPredicate = true;
+      return std::vector<dfa::DFAState::PredPrediction*>();
```

Comment:
Please note that this line of code is wrong. The whole thing should be something like
```cpp
std::vector<dfa::DFAState::PredPrediction*> pairs;
    bool containsPredicate = false;
    for (size_t i = 1; i < altToPred.size(); i++) {
      Ref<SemanticContext> pred = altToPred[i];
  
      // unpredicted is indicated by SemanticContext.NONE
      assert(pred != nullptr);
  
      if (pred != SemanticContext::NONE) {
        containsPredicate = true;
      }
    }
    
    if (!containsPredicate) return pairs;
    
    for (size_t i = 1; i < altToPred.size(); i++) {
      if (ambigAlts.test(i)) {
        pairs.push_back(new dfa::DFAState::PredPrediction(altToPred[i], (int)i)); /* mem-check: managed by the DFAState it will be assigned to after return */
      }
    }
  
    return pairs;
```

Reply:
Doh, Thanks!